core/types/hashing.go

```diff
@@ -56,8 +50,7 @@ func getPooledBuffer(size uint64) ([]byte, *bytes.Buffer, error) {
 
 // rlpHash encodes x and hashes the encoded bytes.
 func rlpHash(x interface{}) (h common.Hash) {
-	sha := hasherPool.Get().(crypto.KeccakState)
-	defer hasherPool.Put(sha)
+	sha := crypto.NewKeccakState()
```

Comment:
This function is used for hashing block. Originally the keccak buffer will be allocated from the pool.

We should somehow keep the syncPool, or use some alternative ways.